### PR TITLE
Complete fizz_buzz_array homework

### PR DESCRIPTION
**Pull request by timsjpark**
[#105165670]

### CODE REVIEW

File: arrays/fizz_buzz_array.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  elsif number % 3 == 0
+    'Fizz'
+  elsif number % 5 == 0
+    'Buzz'
+  else
+    number
+  end
+end
+
+def fizz_buzz(array, index = 0)
```
Comment (jasonnoble) [inline arrays/fizz_buzz_array.rb]: Very creative! :)

File: arrays/fizz_buzz_array.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+
+numbers = (1..100).to_a
```
Comment (jasonnoble) [inline arrays/fizz_buzz_array.rb]: On this one, I'd like to see something like:

    numbers = (1..100).to_a
    fizz_buzz_numbers = numbers._______ { |number| ________(number) }
    puts fizz_buzz_numbers

Comment (timsjpark) [inline arrays/fizz_buzz_array.rb]: I used the `Array.map!` method to modify pre-existing numbers array. I couldn't figure out a way to do it on one line as outlined above without using my original `replace_fizz_buzz` method. But doing so would have resulted in more lines of code and a less elegant product compared to using the pre-existing `.map!` method, in my opinion.

Comment (jasonnoble) [inline arrays/fizz_buzz_array.rb]: I'd still use the `replace_fizz_buzz` method...

    numbers = (1..100).to_a
    fizz_buzz_numbers = numbers.map{ |number|  replace_fizz_buzz(number) }
    puts fizz_buzz_numbers

This way, numbers is still the original array, and fizz_buzz_numbers is a copy of the array with the appropriate substitutions.

Comment (timsjpark) [inline arrays/fizz_buzz_array.rb]: Ah...I didn't realize I could still use the `replace_fizz_buzz` method

**Pull request merged**